docs/docs/api/promise.coroutine.md

```diff
@@ -13,7 +13,9 @@ title: Promise.coroutine
 Promise.coroutine(GeneratorFunction(...arguments) generatorFunction, Object options) -> function
 ```
 
-Returns a function that can use `yield` to yield promises. Control is returned back to the generator when the yielded promise settles. This can lead to less verbose code when doing lots of sequential async calls with minimal processing in between. Requires node.js 0.12+, io.js 1.0+ or Google Chrome 40+.
+In a typical use case, the parameter `generatorFunction` should be a `GeneratorFunction` producing a generator that yields one or more `Promise`s. (Also see the Tips section below.)
```

Comment:
Nit, the generator is the function itself - it returns an iterator. So it's a `GeneratorFunction` but it's not producing a generator function.

Maybe: "In a typical use case, the parameter `generatorFunction` should be a `GeneratorFunction` that can yield zero or more `Promise`s. (Also see the Tips section below.)"?